### ISSUE TITLE
Node.js Foundation Core Technical Committee (CTC) Meeting 2017-05-17

### ISSUE THREAD

**Issue by Trott** [nodejs/CTC#127]
## Time

**UTC Wed 17-May-2017 20:00 (08:00 PM)**:

| Timezone      | Date/Time             |
|---------------|-----------------------|
| US / Pacific  | Wed 17-May-2017 13:00 (01:00 PM) |
| US / Mountain | Wed 17-May-2017 14:00 (02:00 PM) |
| US / Central  | Wed 17-May-2017 15:00 (03:00 PM) |
| US / Eastern  | Wed 17-May-2017 16:00 (04:00 PM) |
| Amsterdam     | Wed 17-May-2017 22:00 (10:00 PM) |
| Moscow        | Wed 17-May-2017 23:00 (11:00 PM) |
| Chennai       | Thu 18-May-2017 01:30 (01:30 AM) |
| Tokyo         | Thu 18-May-2017 05:00 (05:00 AM) |
| Sydney        | Thu 18-May-2017 06:00 (06:00 AM) |

Or in your local time:
* http://www.timeanddate.com/worldclock/fixedtime.html?msg=Node.js+Foundation+Core%20Technical%20Committee%20(CTC)+Meeting+2017-05-17&iso=20170517T20
* or http://www.wolframalpha.com/input/?i=8pm+UTC%2C+May+17%2C+2017+in+local+time

## Links

* **Minutes Google Doc**: <https://docs.google.com/document/d/1GYhQHjCLWDqata2cPh62_lfhW1qDvgbZTGHGCt2Xjs8>
* _Previous Minutes Google Doc: <https://docs.google.com/document/d/1l3HkOFMecOO37FG9YyvFW-ICbCIaw15Ven5xzhUlgs0>_

## Agenda

Extracted from **ctc-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.

### nodejs/node-eps

* Invert dependency between core and readable-stream [#49](https://github.com/nodejs/node-eps/pull/49)

## Invited

* Anna Henningsen @addaleax (CTC)
* Bradley Meck @bmeck (observer/GoDaddy/TC39)
* Ben Noordhuis @bnoordhuis (CTC)
* Сковорода Никита Андреевич @ChALkeR (CTC)
* Colin Ihrig @cjihrig (CTC)
* Evan Lucas @evanlucas (CTC)
* Franziska Hinkelmann @fhinkel (CTC)
* Jeremiah Senkpiel @Fishrock123 (CTC)
* Tracy Hinds @hackygolucky (observer/Node.js Foundation)
* Fedor Indutny @indutny (CTC)
* James M Snell @jasnell (CTC)
* Josh Gavant @joshgav (observer/Microsoft)
* Joyee Cheung @joyeecheung (observer/Alibaba)
* Matteo Collina @mcollina (observer/nearForm)
* Michael Dawson @mhdawson (CTC)
* Julien Gilli @misterdjules (CTC)
* Mikeal Rogers @mikeal (observer/Node.js Foundation)
* Brian White @mscdex (CTC)
* Myles Borins @MylesBorins (CTC)
* Ali Ijaz Sheikh @ofrobots (CTC)
* Jenn Turner @renrutnnej (observer/Node.js Foundation)
* Rod Vagg @rvagg (CTC)
* Shigeki Ohtsu @shigeki (CTC)
* Steven R Loomis @srl295 (observer/IBM/ICU)
* Michaël Zasso @targos (CTC)
* Sakthipriyan Vairamani @thefourtheye (CTC)
* Trevor Norris @trevnorris (CTC)
* Rich Trott @Trott (CTC)

## Notes

The agenda comes from issues labelled with `ctc-agenda` across **all of the repositories in the nodejs org**. Please label any additional issues that should be on the agenda before the meeting starts.

## Joining the meeting

Uberconference; participants should have the link & numbers, contact me if you don't.

## Public participation

We stream our conference call straight to YouTube so anyone can listen to it live, it should start playing at **https://www.youtube.com/c/nodejs+foundation/live** when we turn it on. There's usually a short cat-herding time at the start of the meeting and then occasionally we have some quick private business to attend to before we can start recording & streaming. So be patient and it should show up.

Many of us will be on IRC in #node-dev on Freenode if you'd like to interact, we have a Q/A session scheduled at the end of the meeting if you'd like us to discuss anything in particular. @nodejs/collaborators in particular if there's anything you need from the CTC that's not worth putting on as a separate agenda item, this is a good place for it.

**Comment by Trott** [comment]
Issues labeled `ctc-review`. @nodejs/ctc PTAL:

* [module: remove require('.') with NODE_PATH compat](https://github.com/nodejs/node/pull/3384)
* [http: don't inherit from Object.prototype](https://github.com/nodejs/node/pull/6102)
* [doc: add active, enroll and unenroll](https://github.com/nodejs/node/pull/11736)
* [stream: move prefixed files into internal/streams.](https://github.com/nodejs/node/pull/11957)
* [lib,src: exit process on unhandled promise rejection cleanup](https://github.com/nodejs/node/pull/12010)
* [Who should have access to the security repo ?](https://github.com/nodejs/CTC/issues/96)
* [Why do people need/want to be org owners](https://github.com/nodejs/CTC/issues/97)
* ~~[regression: 3rd party debuggers are incompatible with node8 nighlies](https://github.com/nodejs/node/issues/12364)~~
* ~~[Forming a `GYP` WG](https://github.com/nodejs/CTC/issues/110)~~
* ~~[Uniform way to trigger debugger on first line](https://github.com/nodejs/node/issues/12630)~~
* [Confirmation that the stability definition is correct](https://github.com/nodejs/node/issues/12701)
* [lib,src: eagerly exit process on unhanded promise rejections](https://github.com/nodejs/node/pull/12734)
* [meta: decharter the Docs and Testing Working Groups](https://github.com/nodejs/CTC/pull/121)

**Comment by Trott** [comment]
I'm traveling this week and may not be able to make the meeting, so if someone else could be prepared to run the meeting, that would be awesome.

Related: If you're in or near Portland (Oregon) or Bend (Oregon) and want to meet up this week, get in touch.

**Comment by benjamingr** [comment]
@Fishrock123 I see "lib,src: eagerly exit process on unhanded promise rejections" - would you like to bring it up for discussion in the next CTC meeting?

I'm not sure we're ready yet and I'd like to wait for a month after `util.promisify` lands in a stable (May 30th). On the other hand, maybe presenting some alternatives to the CTC might be a good idea - up to you.

I think "throw on GC, warn on tick" might be a nice improvement made possible with your work.

**Comment by fhinkel** [comment]
I probably won't be able to attend this meeting.

**Comment by joyeecheung** [comment]
It's 4AM in Hangzhou so I probably won't be able to attend.

**Comment by mhdawson** [comment]
@Trott I'm willing to chair the meeting, is there anything specific outside the agenda I should make sure to cover?

**Comment by Trott** [comment]
> @Trott I'm willing to chair the meeting, is there anything specific outside the agenda I should make sure to cover?

@mhdawson Thanks! I don't think there's anything outside the agenda to worry about. I do see that @joshgav removed the `ctc-agenda` label from [the debugger/inspect issue](https://github.com/nodejs/node/issues/12630) so maybe it can be skipped?

**Comment by refack** [comment]
IMHO If there was no interesting discussion about `GYP`'s future during the Collaborator's meeting [Forming a `GYP` WG](https://github.com/nodejs/CTC/issues/110) can be skipped.
I'll do some groundwork in the meanwhile.

**Comment by refack** [comment]
> I do see that @joshgav removed the ctc-agenda label from the debugger/inspect issue so maybe it can be skipped?

Based on the last CTC meeting I believe there's consensus around nodejs/node#12949 so [regression: 3rd party debuggers are incompatible with node8 nighlies](https://github.com/nodejs/node/issues/12364) can also IMHO be skipped.

**Comment by Trott** [comment]
@refack Thanks! FYI, those items were labeled `ctc-review` which means that some CTC input is requested in the GitHub issue/pull request. They normally don't end up being discussed at the meeting. Things labeled `ctc-agenda` end up on the meeting agenda.

That said, it's super-useful to know that they are no longer labeled `ctc-review` because that list of issues is kinda long. The shorter it is, the more attention can be given to items still on it, of course. :-D

**Comment by joshgav** [comment]
As @refack said, I think we have the feedback we need in those issues now so I untagged them to reduce the load on the CTC :)

**Comment by Trott** [comment]
The openssl issue has been closed and the `ctc-agenda` label removed too. That leaves the `readable-stream` issue as the only agenda item at this time. I'll update this issue and the Google Doc.

**Comment by mcollina** [comment]
That's a good time as any, as we cannot ship it until 9, as there is no time, and we are focusing on some old issues. If that is the only reason to have the meeting, we can easily postpone.

**Comment by rvagg** [comment]
OK, if there's nothing of utility to discuss this time then let's can it, so this is a last call -- **anyone got anything they'd like to discuss, private or public, that would justify a meeting this week?**

**Comment by jasnell** [comment]
Nothing significant this week. I'm +1 on skipping as I could use the time back in my calendar today.

**Comment by mhdawson** [comment]
Sames goes for me.  Happy to skip the meeting if we have no issues to discuss.

**Comment by ChALkeR** [comment]
I wanted to mention something in private, but that could be postponed until the next week or moved to email, so I'm good with skipping.

**Comment by addaleax** [comment]
I guess I would have liked to do my thing where I get annoying and ping everyone about the state of our collaborator nominations (https://github.com/nodejs/node/issues/12646) :smile: But well, three of them have been nominated for long enough so that everybody has had plenty of time to voice objections, and I haven’t seen any, so I guess I can send out emails later anyway.  ¯\\\_(ツ)_/¯

**Comment by indutny** [comment]
Adding this item https://github.com/nodejs/node/pull/13005 to ctc-agenda. I'm not yet sure if I'll be able to make it today, but would appreciate if some (at least preliminary) discussion could be made with regards to this API addition.

Motivation for the change is making Agent API more flexible for external use.

**Comment by mhdawson** [comment]
@indutny looking at the comments on the issue, it does not seem like there would be a meeting this week.  Since you are not sure if you can make it how about it be on the agenda for next week ?

**Comment by indutny** [comment]
Sounds good!

**Comment by mhdawson** [comment]
Unless somebody speaks up in the next 10 mins, I think we should officially cancel the meeting for this week.

**Comment by mhdawson** [comment]
Ok, I think we can declare it cancelled, closing the issue.
